sktime/utils/mlflow_sktime.py

```diff
@@ -0,0 +1,534 @@
+#!/usr/bin/env python3 -u
+# -*- coding: utf-8 -*-
+"""The ``mlflow_sktime`` module provides an MLflow API for ``sktime`` models.
+
+This module exports ``sktime`` models in the following formats:
+
+sktime (native) format
+    This is the main flavor that can be loaded back into sktime, which relies on pickle
+    internally to serialize a model.
+mlflow.pyfunc
+    Produced for use by generic pyfunc-based deployment tools and batch inference.
+
+    The ``pyfunc`` flavor of the model supports sktime predict methods ``predict``,
+    ``predict_interval``, ``predict_quantiles`` and ``predict_var`` (``predict_proba``
+    and ``predict_residuals`` are currently not supported).
+"""
+
+__author__ = ["benjaminbluhm"]
+__all__ = [
+    "get_default_pip_requirements",
+    "get_default_conda_env",
+    "save_model",
+    "log_model",
+    "load_model",
+]
+
+import logging
+import os
+import pickle
+
+import pandas as pd
+import yaml
+from mlflow import pyfunc  # noqa: F401
+from mlflow.tracking._model_registry import DEFAULT_AWAIT_MAX_SLEEP_SECONDS
+from mlflow.utils.docstring_utils import LOG_MODEL_PARAM_DOCS, format_docstring
+
+import sktime
+from sktime import utils
+from sktime.utils.validation._dependencies import _check_soft_dependencies
+
+_check_soft_dependencies("mlflow", severity="warning")
+
+FLAVOR_NAME = "mlflow_sktime"
+_MODEL_BINARY_KEY = "data"
+_MODEL_BINARY_FILE_NAME = "model.skt"
+_MODEL_TYPE_KEY = "model_type"
+PREDICT_METHODS = ["predict", "predict_interval", "predict_quantiles", "predict_var"]
```

Comment:
Add this point it is not trivial to add these methods to the `pyfunc` wrapper class as one would need to allow passing `y` and `X`, but one can only pass single DataFrame to `pyfunc.predict`. Perhaps it could be done by concatenating `y` and `X` but at the very least it would make the inference logic much more involved. Would suggest to keep this in mind for a potential future release.

In any case one can load the model in native sktime format via `mlflow_sktime.load_model(model_uri=model_path)` and then call `update_predict`, `update_predict_single` etc.